clik/external/hal_cpu/include/cpu_hal.h

```diff
@@ -48,6 +48,10 @@ class cpu_hal final : public hal::hal_device_t {
   ~cpu_hal();
 #endif
 
+  // Set up the hal device info - this is done as a static so that other classes such
+  // as hal_client can set up the desired information directly
+  static void setup_cpu_hal_device_info(hal::hal_device_info_t &hal_device_info);
```

Comment:
Agreed that using a static helper function for this is a good idea, but can it return `hal_device_info_t` by value so it is clear it does not rely on the caller set up anything in it in advance?

Reply:
sure